src/code/z_demo.c

```diff
@@ -333,202 +367,193 @@ void Cutscene_Command_Misc(PlayState* play, CutsceneContext* csCtx, CsCmdBase* c
+            play->envCtx.lightSettingOverride = cmd->settingPlusOne - 1;
             play->envCtx.lightBlend = 1.0f;
         } else {
             play->envCtx.lightSettingOverride = 0xFF;
         }
     }
 }
 
-// Command 0x12C: Plays a sequence (Background music or Fanfare)
-void Cutscene_Command_PlaySequence(PlayState* play, CutsceneContext* csCtx, CsCmdSequenceChange* cmd) {
-    if (csCtx->frames == cmd->startFrame) {
-        Audio_PlaySequenceInCutscene(cmd->sequence - 1);
+void CutsceneCmd_StartSequence(PlayState* play, CutsceneContext* csCtx, CsCmdStartSeq* cmd) {
+    if (csCtx->curFrame == cmd->startFrame) {
+        Audio_PlaySequenceInCutscene(cmd->seqIdPlusOne - 1);
     }
 }
 
-// Command 0x12D: Stops a sequence (Background music or Fanfare)
-void Cutscene_Command_StopSequence(PlayState* play, CutsceneContext* csCtx, CsCmdSequenceChange* cmd) {
-    if ((csCtx->frames >= cmd->startFrame) && (cmd->endFrame >= csCtx->frames)) {
-        Audio_StopSequenceInCutscene(cmd->sequence - 1);
+void CutsceneCmd_StopSequence(PlayState* play, CutsceneContext* csCtx, CsCmdStopSeq* cmd) {
+    if ((csCtx->curFrame >= cmd->startFrame) && (csCtx->curFrame <= cmd->endFrame)) {
+        Audio_StopSequenceInCutscene(cmd->seqIdPlusOne - 1);
     }
 }
 
-// Command 0x9C: Fade a sequence (Background music or Fanfare) over duration
-void Cutscene_Command_FadeSequence(PlayState* play, CutsceneContext* csCtx, CsCmdSequenceFade* cmd) {
-    if ((csCtx->frames == cmd->startFrame) && (csCtx->frames < cmd->endFrame)) {
-        u8 fadeTimer = cmd->endFrame - cmd->startFrame;
+void CutsceneCmd_FadeOutSequence(PlayState* play, CutsceneContext* csCtx, CsCmdFadeOutSeq* cmd) {
+    if ((csCtx->curFrame == cmd->startFrame) && (csCtx->curFrame < cmd->endFrame)) {
+        u8 fadeOutDuration = cmd->endFrame - cmd->startFrame;
 
-        if (cmd->type == 2) {
-            Audio_QueueSeqCmd((fadeTimer << 0x10) | 0x110000FF);
-        } else {
-            Audio_QueueSeqCmd((fadeTimer << 0x10) | NA_BGM_STOP);
+        if (cmd->seqPlayer == CS_FADE_OUT_FANFARE) {
+            Audio_QueueSeqCmd((fadeOutDuration << 0x10) | 0x110000FF);
+        } else { // CS_FADE_OUT_BGM_MAIN
+            Audio_QueueSeqCmd((fadeOutDuration << 0x10) | NA_BGM_STOP);
         }
     }
 }
 
-// Command 0x12E: Play Ambience sequence
-void Cutscene_Command_PlayAmbienceSequence(PlayState* play, CutsceneContext* csCtx, CsCmdBase* cmd) {
-    if (csCtx->frames == cmd->startFrame) {
+void CutsceneCmd_StartAmbience(PlayState* play, CutsceneContext* csCtx, CsCmdStartAmbience* cmd) {
+    if (csCtx->curFrame == cmd->startFrame) {
         Audio_PlayAmbience(play->sequenceCtx.ambienceId);
     }
 }
 
-// Command 0x130:
-void func_800EAD48(PlayState* play, CutsceneContext* csCtx, CsCmdBase* cmd) {
-    if (csCtx->frames == cmd->startFrame) {
+void Cutscene_SetSfxReverbIndexTo2(PlayState* play, CutsceneContext* csCtx, CsCmdSfxReverbIndexTo2* cmd) {
+    if (csCtx->curFrame == cmd->startFrame) {
```

Comment:
i dont know exactly what this is, but i imagine the name could probably be less verbose than "reverb index". like, I dont think the user cares that its an index into something audio side.

but if some more docs audio side would help then its fine for now

Reply:
This one is annoying, and even with full knowledge on what reverb index is, I can't think of any way to simplify this name.

To describe reverb index, there are reverb settings that are provided with audio specifications (i.e. a set of data for each audio session, roughly aligning with 1 session/scene, just like PlayState).

However, there are usually 2-3 different reverb settings you can assign to each audio seqPlayer/channel. The reverb index is the index of these reverb settings that you can choose from. Bgm/Fanfares are usually reverbIndex == 0:
https://github.com/zeldaret/mm/blob/ab8d34b8dc2ccd28ae4224d8500decff0fa25a1c/src/code/audio/audio_init_params.c#L30-L31

And sfxs are usually split by channel between reverbIndex 0/1/2.

There are two commands to toggle the reverbIndex of all sfx channels between 1 and 2 i.e. `Cutscene_SetSfxReverbIndexTo1` and `Cutscene_SetSfxReverbIndexTo2`

Something like `Cutscene_SetReverb1` and ``Cutscene_SetReverb2` is even less clear. Really, it could be `Cutscene_SetSfxReverbSettingsIndexTo1` and ``Cutscene_SetSfxReverbSettingsIndexTo2` if `reverbIndex` -> `reverbSettingsIndex` but idk. I don't see `reverbIndex` being shortened, since `reverb` means something else.

Even the OoT Audio data PR could only shorten it to remove vowels: `rvrbIdx`, which would not look good in the function name/misc commands

Reply:
yeah this is tough. I think we will need a higher level term for this "index". like "mode" or something. 
or the "settings" can be "parameters", and the setting is a group of these parameters

something to discuss audio side anyway, and if that ever changes this can update to reflect that

Reply:
reverb preset, which defines these group of settings, idk

Reply:
That's true, I could see `reverbIndex` -> `reverbPresetId` or `reverbSettingsId` or something. I'll make a mental note to sync these cutscene commands with that name change when that discussion arrives